### ISSUE TITLE
Clean up the initial setup

### ISSUE THREAD

**Issue by EinarElen** [EinarElen/prototype-geometry#1]
There are parts of the geometry that we would like to either remove or make easier to read before we send anything to the mainline LDMX-sw. 

- [x] Rename the constants file to remove reference to ears ed25e9a6295450a24b635aa0235cbd327ba2b05d
- [x] Group the four scintillator volume files into one dd4e6d61359703751b1209923f9c46fb25a0e160
- [x] Remove references to things that aren't part of the geometry such as the Side HCal  dd4e6d61359703751b1209923f9c46fb25a0e160
- [x] Change loop indices to a single index using the from-parameter 6f9efa40211aa1051094f959afaac8471b16bb51
- [x] Apply snake_case to all variables we control 
    - [x] numLayers 7ba784a5632d01fdacf8a76f85e77b23126e703a, 005f20854a35fa7287a8c5202e9d37345984f653, e0abf3e00f2a2a0ba3f7df6a2be32876970bb934
    - [x] scint_Front/Back 9bc8c2866cffa228001a0b3ab8d0dc981488d2dd
    - [x] ScintBox d34eb208c168a6852be5e555ef0987e8ded3b7a5 
    - [x] Other "Box" variables ced2cdd3b3b465a2d80d28a95083871fb2e50e01
- [x] Remove unused VisAttributes 9bfbc34775d1ccff43cfcb1678b637729abfefe1
- [x] Remove identity variable a5d5afd08af6d85fabf51fe70b628a3072c2a2db
- [x] Move the Hcal volume into a dedicated file 91e64993f7625b0f662e6b3b7b4f965b392a6d62
- [x] Merge the HCal logical volumes into one file 637d535e5a9c64da3e4d644a8e17ae29e78f72b4
- [ ] Remove magic constants and define variables in terms of each other
    - [x] 400/600 mm 19bccb1762b87713ecd6aba471523357850aa82a
    - [x] numLayers family of variables 681dc687ed0942c4ae98d7dd2478ffcfc1b77c19
    - [x] absorber x/y position derived from absorber width 6ccb0c104f723784998ea48c798f91b4fd2259d2
- [x] Move world solid to detector.gdml e6d3862669e769a25158dee8fa37f1215adaa233
- [ ] Remove messy abbreviations 
    - [x] V/H/vert/hori 681dc687ed0942c4ae98d7dd2478ffcfc1b77c19
    - [x] scint_thick ff46eef434f62ffb97a5eca491f8af33411b6440
    - [x] air_thick 6924b4a8c16840d43d5f89eef731b1aef6af9369
    - [x] layer_thick b507f91eb9232cc82b4c4c95b11c1c09e4f9a6d6
- [x] Change the DoubleLayers vocabulary 681dc687ed0942c4ae98d7dd2478ffcfc1b77c19
- [ ] Find a naming scheme for the fron/back regions that doesn't conflict with LDMX-sw 
- [x] Move calculation of the position of the first scintillator layers to a variable instead of doing it in the loop to create physvols  21843a5cc05f14fbb2bf9b2a9c703d42f982b2f4
- [ ] #7
- [ ] Prefix the prototype-dx/dy 
- [ ] Use the same scheme for prefixing things everywhere

**Comment by EinarElen** [comment]
> > * [ ]     We are very inconsistent in variable naming style. Some variables use
> >   camelCase such as `numLayersFront`, some use snakecase such as `scint_thick`,
> >   some use a weird mixture like `scint_FrontH_x`, and some use PascalCase such
> >   as `CalorimeterRegion`.
> 
> That is bad. We should use just one. Although det-v12 (the one I have at least) is also very inconsistent, so the apple doesn't fall far from the tree. Which standard we end up using is up for discussion.
> 
  A problem with choosing a consistent naming style is that we don't control all  of the names, some are standard Geant4 names and some seem to be used by LDMX-sw internally.

  I would propose that we pick one naming style for everything that we control, which means that we will know that anything which doesn't follow this style is a name that we actually don't own. Since neither Geant nor LDMX-sw are using `snake_case`, I propose we use `snake_case` for all of our variables.


> > * [ ]     The width/height of the prototype, `dx` and `dy`, are both set to **3000 mm**.
> >   Since the bars are **2000 mm**, shouldnt this be the same? Or do we have a
> >   reason for the prototype volume being larger than
> 
> I did just kill 2 hours just to discover the root of the mysterious error was placing things outside the prototype volume. But I will admit it is more logical to have a box be the exact size of the things inside. But why do we even call it dx and dy??? When people put it in for loops, I immediately think there's some steps being taken à la integral/derivative style. It's really counterintuitive.
> 
> > * [ ]     We are inconsistent in calling things thick or thickness
> >   Thick is a horrible abbreviation of thickness since thick is a word on its own - usually an adjective - and is separate from thickness, which is always a noun.
> >   
> >   * In general, we should decide if we write out things in full or use
> >     abbreviations everywhere. Current abbreviations we use are
> >     
> >     * [ ]   thick, thickness
> >     * [ ]   pos, position
> >     * [ ]   dim, dimension
> >     * [ ]   scint, scintillator
> >     * [ ]   V, vert, vertical
> >     * [ ]   H, hori, horizontal
> 
> I suppose using abbreviations actually makes the code more readable in these obvious cases. Except for thick, since that abbreviation is pure bad.
> 
> > * Should we call these things thickness/width? I suspect it is good to use the
> >   same nomenclature as the main LDMX-sw detector descriptions, but in
> >   principle I don’t think these are the most descriptive terms. At least I
> >   have to stop myself to think about which one means dx & dy and which one
> >   means dz.
> 
> I do admit I don't like the term thickness. Not just because of abbreviation, but because it's also round and sort of implies both x and z directions. I advocate for killing it if we can.
> 
> > * [ ]     The back region layers are currently called “Double layers”. I find this very
> >   confusing. It is called this way since the back region has the same number of
> >   vertical and horizontal layers so we only need one loop limit, compared with
> >   the front region where we need to loop once to four layers for the vertical
> >   layers but up to five for the horizontal layers. Regardless, there has to be a
> >   better name for this or just have two variables for the back region similar to
> >   what we have for the front region even though it would imply a bit of
> >   repetition
> 
> Yeah the term double layers has to go away. I approve of scrapping it.
> 
> > * [ ]     Calling the layers front/back can be a bit confusing since the entire
> >   prototype is technically part of the “back” section. Maybe we should decide on
> >   a different naming scheme to use here to be clear? Large/small layers?
> >   Something else?
> 
> Front/back makes sense to me. And v12's gdml doesn't have anything called back. Large/small would be bad because it's just as tall and thick, simply with fewer bars. Can't think of a better naming scheme.
> 

> > * [ ]   The width (height), i.e. Δ xlayer (Δ ylayer) of the vertical
> >   (horizontal) layers could have two variables instead of repeating 400/600
> >   mm. Alternatively, we could have variables to explicitly represent the width
> >   of a scintillator (50 mm) and the number of bars in the front (back) regions
> >   and calculate the width (height) directly from these.
> 
> I don't understand what you mean.
>
  Currently, we are definining variables like `scint_FrontV_x = 400`, `scint_FrontV_y = scint_bar_length`. I would suggest that we rather than just entering constants like "400" create an expression that actually shows where this number is derived from.

  In this case we would have a `numBarsFront = 8` and `scint_width = 50` such that we could define `scint_FrontV_x = numBarsFront * scint_width`.

> > * [ ]     Do we want the prototype volume to start in the middle of the world volume
> >   rather than having the first layer to start at z=0?
> 
> Actually, I would prefer starting at 0, yes. But the trigger is slightly before the absorber, so what would we do about that?

We would have to decide either if we start with the trigger scintillator at z=0 or before. I would probably prefer keeping the depth entirely positive so we would put the trigger scintillator at 0. But we can leave this as is for now


> > * [ ]     It would be nice to go through the visualization options to create nicer
> >   looking visuals by default. For example, the different parts could have
> >   different colours by default instead of all being white
> 
> Sure, I guess.
>
  Actually, it seems like the visualization stuff that is in there doesn't actually work with the way we are doing visualization but is rather used by LDMX-sw internally. So we have no reason to mess with it in the GDML-code and have to simply do this in the Geant4 visualization macros explicitly...

**Comment by PeterGy** [comment]
> I would propose that we pick one naming style for everything that we control, which means that we will know that anything which doesn't follow this style is a name that we actually don't own. Since neither Geant nor LDMX-sw are using `snake_case`, I propose we use `snake_case` for all of our variables.

Sure, we shall use snake_case

 

> However, dx/dy/dz are kind of bad names regardless because it isn't clear what they are the dimensions of. I think they could use a prefix to make this more clear. That should also hopefully help with making them look less like derivatives/integrals.

I've only seen Δx/Δy be used in the context of classical and quantum uncertainty, as well as non-infinitesimal steps in math courses. I suppose the logic in defining the length of an object that way is that you have its 2 end point coordinates x1 and x2, and length = Δx = x2-x1. Is that right?
Also, we definitely need to add a prefix to the dx dy dz thing that currently exists alone.



> I would actually disagree here. Some of them could arguably be roughly the same amount of effort to parse when you are reading through the code. I would say that pos/dim/scint probably fall into these categories.

More elaborately, I think pos/dim/scint are easier to parse than their longer counterparts. h/v hor/vert are harder to parse than their longer counterparts.

> The point isn't if the names are obvious in isolation, but if they remain obvious in context. I would argue that the vert/hori and especially V/H make the placement code unnecessarily hard to parse. At least I always have to stop myself to look and make sure that I'm looking at the right block.

Yup

> So I would propose here that we either don't use abbreviations at all for these things or that we can abbreviate pos/dim/scint but not the other three.

Abbreviate pos/dim/scint but not the other three

> One place where we could introduce an abbreviation would be for the `hadron_calorimeter_pos` variable. `hcal_pos` would probably be more clear.

Yup


> > I do admit I don't like the term thickness. Not just because of abbreviation, but because it's also round and sort of implies both x and z directions. I advocate for killing it if we can.
> 
> Yeah, I agree. I am struggling a bit with finding a better name however so maybe we will just have to stick with it for now.

Maybe we could use (dx,dy,dz) for non rotating objects (like triggers and absorbers) and (length,width,dz) for the rotating scintillator bars. The idea being that since the third coordinate will always be in z, we can call it dz, while otherwise we'd have to call it thickness, which is bad. https://github.com/EinarElen/prototype-geometry/issues/7#issuecomment-884127389


> I would actually suggest here that we just use the same structure as we do for the front region. I.e. there is one variable for the vertical layers and one variable for the horizontal layers. Technically, this is introducing some repetition but I really think it would be more clear than the current way.

Good idea 

> So we can't just think about the GDML-files, since the names we decide to use here are also used in the reconstruction code. And here, the name "Back" is already occupied by the section-number, so "Back" really doesn't work well.
> 
> I can't think of a good name at the moment however, but we should definitely try and think of something.

Sure.


> In this case we would have a `numBarsFront = 8` and `scint_width = 50` such that we could define `scint_FrontV_x = numBarsFront * scint_width`.

Good idea.


> We would have to decide either if we start with the trigger scintillator at z=0 or before. I would probably prefer keeping the depth entirely positive so we would put the trigger scintillator at 0. But we can leave this as is for now

I have a feeling the trigger scintillator may be moved around a bit. We could make the hcal part start at 0 and let the triggers sit in the negatives. Or just leave it as is now.

> Actually, it seems like the visualization stuff that is in there doesn't actually work with the way we are doing visualization but is rather used by LDMX-sw internally. So we have no reason to mess with it in the GDML-code and have to simply do this in the Geant4 visualization macros explicitly...

I'm fine without visualisation commands inside the .gdml files with my current visualisation setup. And I might try to upgrade my visualiser now that I managed to get a MobaXterm display to work with my wsl, and I don't yet know if or how changing gdml code would help that visualisation.
